test/functional/page_objects/common_page.ts

```diff
@@ -164,116 +202,84 @@ export function CommonPageProvider({ getService, getPageObjects }) {
       } else {
         appUrl = getUrl.noAuth(config.get('servers.kibana'), {
           pathname: `${basePath}/app/${appName}`,
-          hash
+          hash,
         });
       }
 
       log.debug('navigating to ' + appName + ' url: ' + appUrl);
 
-      function navigateTo(url) {
-        return retry.try(function () {
+      await retry.tryForTime(defaultTryTimeout * 2, async () => {
+        let lastUrl = await retry.try(async () => {
           // since we're using hash URLs, always reload first to force re-render
-          return kibanaServer.uiSettings.getDefaultIndex()
```

Comment:
I removed this line since we didn't use the return result. Not sure what was the purpose?

Reply:
There was code here at one time that checked if the default index was set when we tried to navigate to Discover, Visualize, Dashboard (but not Management).  If you didn't have it set, it would set it to `logstash-*` and log a warning message.  Maybe we fixed any tests which weren't setting the default index, in which case that code isn't needed any more.